Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -171,392 +177,470 @@ NonnullRefPtr<CSSStyleSheet> Parser::parse_as_stylesheet(TokenStream<T>& tokens)
+                simple_selector.pseudo_element = Selector::SimpleSelector::PseudoElement::Before;
+            } else if (pseudo_name.equals_ignoring_case("after")) {
+                // See :before
+                simple_selector.type = Selector::SimpleSelector::Type::PseudoElement;
+                simple_selector.pseudo_element = Selector::SimpleSelector::PseudoElement::After;
+            } else if (pseudo_name.equals_ignoring_case("first-line")) {
+                // See :before
+                simple_selector.type = Selector::SimpleSelector::Type::PseudoElement;
+                simple_selector.pseudo_element = Selector::SimpleSelector::PseudoElement::FirstLine;
+            } else if (pseudo_name.equals_ignoring_case("first-letter")) {
+                // See :before
+                simple_selector.type = Selector::SimpleSelector::Type::PseudoElement;
+                simple_selector.pseudo_element = Selector::SimpleSelector::PseudoElement::FirstLetter;
+            } else {
+                dbgln_if(CSS_PARSER_DEBUG, "Unknown pseudo class: '{}'", pseudo_name);
+                return SelectorParsingResult::SyntaxError;
+            }
 
-        return Selector::ComplexSelector { relation, move(simple_selectors) };
-    };
+            return simple_selector;
 
-    for (;;) {
-        auto current_value = tokens.peek_token();
-        if (current_value.is(Token::Type::EndOfFile))
-            break;
+        } else if (pseudo_class_token.is_function()) {
 
-        auto complex = parse_complex_selector();
-        if (complex.has_value())
-            selectors.append(complex.value());
-    }
+            auto& pseudo_function = pseudo_class_token.function();
+            if (pseudo_function.name().equals_ignoring_case("nth-child")) {
+                simple_selector.pseudo_class.type = Selector::SimpleSelector::PseudoClass::Type::NthChild;
+                auto function_values = TokenStream<StyleComponentValueRule>(pseudo_function.values());
+                auto nth_child_pattern = parse_a_n_plus_b_pattern(function_values);
+                if (nth_child_pattern.has_value()) {
+                    simple_selector.pseudo_class.nth_child_pattern = nth_child_pattern.value();
+                } else {
+                    dbgln_if(CSS_PARSER_DEBUG, "!!! Invalid nth-child format");
+                    return SelectorParsingResult::SyntaxError;
+                }
+            } else if (pseudo_function.name().equals_ignoring_case("nth-last-child")) {
+                simple_selector.pseudo_class.type = Selector::SimpleSelector::PseudoClass::Type::NthLastChild;
+                auto function_values = TokenStream<StyleComponentValueRule>(pseudo_function.values());
+                auto nth_child_pattern = parse_a_n_plus_b_pattern(function_values);
+                if (nth_child_pattern.has_value()) {
+                    dbgln("+++ Got An+B pattern: '{}'", nth_child_pattern.value().to_string());
```

Comment:
I think this is a leftover?
It's not present in the :nth-child case and every other dbgln is coupled to the CSS_PARSER_DEBUG

Reply:
Oops! I thought I'd removed them all. Thanks for catching it. :^)